changelog/gc_registry.dd

```diff
@@ -0,0 +1,33 @@
+User supplied garbage collectors can now be linked with the runtime
+
+A GC registry has been implemented that allows to add
+garbage collector implementations by just linking them into
+the binary. To do so add a function that is executed before the
+D runtime initialization using `pragma(crt_constructor)`:
+
+---
+extern (C) pragma(crt_constructor) void registerMyGC()
+{
+    registerGCFactory("mygc", &createMyGC);
+}
+
+GC createMyGC()
+{
+	__gshared ubyte[__traits(classInstanceSize, MyGC)] buf;
```

Comment:
Probably good to add an alignment here see e.g. https://github.com/dlang/druntime/pull/2411

Reply:
Unfortunately, classInstanceAlignment doesn't seem good enough (and isn't yet available), so I opted for aligning to pointer size.
I also used core.lifetime.emplace now.